src/analysis/processing/qgsalgorithmstdbscanclustering.cpp

```diff
@@ -0,0 +1,339 @@
+}
+
+QString QgsStDbscanClusteringAlgorithm::group() const
+{
+  return QObject::tr( "Vector analysis" );
+}
+
+QString QgsStDbscanClusteringAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectoranalysis" );
+}
+
+void QgsStDbscanClusteringAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterFeatureSource( QStringLiteral( "INPUT" ),
+                QObject::tr( "Input layer" ), QList< int >() << QgsProcessing::TypeVectorPoint ) );
+  addParameter( new QgsProcessingParameterField( QStringLiteral( "DATETIME_FIELD" ),
+                QObject::tr( "Date/time field" ), QVariant(),
+                QStringLiteral( "INPUT" ), QgsProcessingParameterField::DateTime, false, false ) );
+
+  addParameter( new QgsProcessingParameterNumber( QStringLiteral( "MIN_SIZE" ), QObject::tr( "Minimum cluster size" ),
+                QgsProcessingParameterNumber::Integer, 5, false, 1 ) );
+  addParameter( new QgsProcessingParameterDistance( QStringLiteral( "EPS1" ),
+                QObject::tr( "Maximum distance between clustered points" ), 1, QStringLiteral( "INPUT" ), false, 0 ) );
+  addParameter( new QgsProcessingParameterNumber( QStringLiteral( "EPS2" ),
+                QObject::tr( "Maximum date/time interval (in days unit) between clustered points" ), QgsProcessingParameterNumber::Double, 1, false, 0 ) );
+
+  auto dbscanStarParam = std::make_unique<QgsProcessingParameterBoolean>( QStringLiteral( "DBSCAN*" ),
+                         QObject::tr( "Treat border points as noise (DBSCAN*)" ), false, true );
+  dbscanStarParam->setFlags( dbscanStarParam->flags() | QgsProcessingParameterDefinition::FlagAdvanced );
+  addParameter( dbscanStarParam.release() );
+
+  auto fieldNameParam = std::make_unique<QgsProcessingParameterString>( QStringLiteral( "FIELD_NAME" ),
+                        QObject::tr( "Cluster field name" ), QStringLiteral( "CLUSTER_ID" ) );
+  fieldNameParam->setFlags( fieldNameParam->flags() | QgsProcessingParameterDefinition::FlagAdvanced );
+  addParameter( fieldNameParam.release() );
+  auto sizeFieldNameParam = std::make_unique<QgsProcessingParameterString>( QStringLiteral( "SIZE_FIELD_NAME" ),
+                            QObject::tr( "Cluster size field name" ), QStringLiteral( "CLUSTER_SIZE" ) );
+  sizeFieldNameParam->setFlags( sizeFieldNameParam->flags() | QgsProcessingParameterDefinition::FlagAdvanced );
+  addParameter( sizeFieldNameParam.release() );
+
+  addParameter( new QgsProcessingParameterFeatureSink( QStringLiteral( "OUTPUT" ), QObject::tr( "Clusters" ), QgsProcessing::TypeVectorPoint ) );
+
+  addOutput( new QgsProcessingOutputNumber( QStringLiteral( "NUM_CLUSTERS" ), QObject::tr( "Number of clusters" ) ) );
+}
+
+QString QgsStDbscanClusteringAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "Clusters point features based on a 2D implementation of spatiotemporal density-based clustering of applications with noise (ST-DBSCAN) algorithm." );
```

Comment:
Afaics this algorithm is based on https://www.sciencedirect.com/science/article/pii/S0169023X06000218 -- correct? If so, would be great to reference that paper in the short help string!

Reply:
Or even better, reference all the papers from the readme here: https://github.com/eren-ck/st_dbscan